src/esw/can_node/main.cpp

```diff
@@ -0,0 +1,26 @@
+#include "CanNode.hpp"
+#include <mrover/CAN.h>
+#include <ros/ros.h>
+
+void handleMessage(const mrover::CAN::ConstPtr& msg);
+
+CanNode node(false);
+
+int main(int argc, char** argv) {
+    ros::init(argc, argv, "can_node");
+    ros::NodeHandle nh;
+
+    ros::Subscriber CanSubscriber = nh.subscribe<mrover::CAN>("can_requests", 1, handleMessage);
+
+    ros::spin();
+
+    return 0;
+}
+
+void handleMessage(const mrover::CAN::ConstPtr& msg) {
+    node.set_bus(msg->bus);
+    node.set_frame_id(msg->message_id);
+    node.set_frame_data(std::span<std::byte>(reinterpret_cast<std::byte*>(msg->data.data()), msg->data.size()));
```

Comment:
Yeah... this is unfortunately an annoying system since `std::byte` is modern and ROS is old

Reply:
If the types matched you could just pass the `msg->data` directly which is the whole point of `std::span`...

Reply:
Once you change the signature to take in a `std::span<const std::byte> data` you can do:

`node.set_frame_data({reinterpret_cast<std::byte const*>(msg->data.data()), msg->data.size()});`

Reply:
Note the const qualifier. `set_frame_data` should not be able to mutate the byte buffer it is given.

Reply:
Also `handleMessage` should probably be in the CANNode class. You can then pass `this` as a fourth argument to subscribe.